### DISCUSSION TITLE
Documentation for block types?

### DISCUSSION THREAD

**Discussion by HydrogenC** [huggingface/diffusers#7664] [Q&A]
I failed to find any documentation about the block types (defined in `unet_2d_blocks.py`) on the official site, is there anywhere to find any docs for them?  
What I am trying to figure out is the difference between `UpBlock2d` and `UpDecoderBlock2d` (or `DownBlock2d` and `DownDecoderBlock2d`). I dived into the code and found them implemented in very similar ways, especially with `resnet_time_scale_shift` defined to false. So why exactly are their differences?

**Comment by yiyixuxu** [comment]
cc @DN6 
I think we should consolidate if it is indeed the same
